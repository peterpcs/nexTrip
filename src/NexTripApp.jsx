import React from 'react'
import App from './App'
import store from './store'
import { Provider } from 'react-redux'
import FindByRoute from './components/directions/ByRoute'
import FindByStop from './components/directions/ByStop'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'

export default function NexTripApp() {
  return (
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<App />}>
            <Route path="route" element={<FindByRoute />} />
            <Route path="stop" element={<FindByStop />} />
            <Route index element={<FindByRoute />} />
          </Route>
        </Routes>
      </Router>
    </Provider>
  )
}
