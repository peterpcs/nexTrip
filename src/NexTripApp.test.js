import * as React from 'react'
import NexTripApp from './NexTripApp'
import { rest } from 'msw'
import { setupServer } from 'msw/node'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from './test-utils'
import renderer from 'react-test-renderer'

const route = {
  route_id: '901',
  agency_id: 0,
  route_label: 'METRO',
}
const routes = [
  {
    ...route,
  },
]

const handlers = [
  rest.get('https://svc.metrotransit.org/nextripv2/routes', (req, res, ctx) => {
    return res(ctx.json(routes))
  }),
]

const server = setupServer(...handlers)

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())

test('renders nexTrip app', () => {
  render(<NexTripApp />)
  expect(screen.getByText(/NexTrip/i)).toBeInTheDocument()
  expect(screen.getByText(/By route/i)).toBeInTheDocument()
  expect(screen.getByText(/By stop/i)).toBeInTheDocument()
})

test('renders correctly', () => {
  const tree = renderer.create(<NexTripApp />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('expect by-route tab to be active by default', () => {
  render(<NexTripApp />)

  const routeTabElement = screen.getByText(/By route/i)
  expect(routeTabElement).toBeInTheDocument()
  expect(routeTabElement.classList.contains('active')).toBe(true)
})

test('tab should be active on click', () => {
  render(<NexTripApp />)

  const stopTabElement = screen.getByText(/By stop/i)
  expect(stopTabElement).toBeInTheDocument()
  userEvent.click(stopTabElement)
  expect(stopTabElement.classList.contains('active')).toBe(true)

  const routeTabElement = screen.getByText(/By route/i)
  expect(routeTabElement).toBeInTheDocument()
  userEvent.click(routeTabElement)
  expect(routeTabElement.classList.contains('active')).toBe(true)
})
