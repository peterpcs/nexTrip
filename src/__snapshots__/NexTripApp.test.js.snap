// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="nextTrip_title"
    >
      NexTrip
    </h1>
  </header>
  <main>
    <div
      className="MuiContainer-root MuiContainer-maxWidth1140px css-1itnxbn-MuiContainer-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1b8vvdz-MuiTypography-root"
      >
        Real-time Departures
      </div>
      <div
        className="MuiBox-root css-1kax7eu"
      >
        <a
          aria-current="page"
          className="nextrip-tab active"
          href="/route"
          onClick={[Function]}
        >
          By route
        </a>
        <a
          className="nextrip-tab"
          href="/stop"
          onClick={[Function]}
        >
          By stop #
        </a>
      </div>
      <div
        className="MuiBox-root css-8fkoyy"
      >
        <div
          className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth css-1pkfvy1-MuiFormControl-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              id="route-select-label"
            >
              Select route
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              data-testid="route"
              onClick={[Function]}
              variant="outlined"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="route route-select"
                className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="route-select"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="notranslate"
                >
                  â€‹
                </span>
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                disabled={false}
                name="route"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value=""
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                    Select route
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
